Suppose the IQ in a certian population  is normally distrbuted with a mean of MU(µ) = 100 and SD of σ = 15

A researcher wants to Know if a new drug affects IQ levels, so he recruits 20 patients to try it and records their IQ levels.

The following code shows how to perform a one sample Z-test in python to determine if the new drug causes a significant difference in IQ levels.

In [8]:
from statsmodels.stats.weightstats import ztest as ztest

# enter IQ levels for 20 patients.
data = [88, 92, 94, 94, 96, 97, 97, 97, 99, 99, 105, 109, 109, 109, 110, 112, 112, 113, 114, 115]

In [9]:
ztest(data,value = 100)
#first value output is z-test, 2 is P-value.
# first value is actual z-test value. 2 is to just for comparing the p-value and reject or accept H0 .

(1.5976240527147705, 0.1101266701438426)

here 0.11 > then 0.05 so we reject h0


if pm val is less then signeficance value we reject the H0.